src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Build/Diagnostics/IncrementalBuildFailureDetector.ProjectChecker.cs

```diff
@@ -24,7 +24,7 @@ internal sealed partial class IncrementalBuildFailureDetector
         private sealed class ProjectChecker : IProjectChecker
```

Comment:
Couldn't there be some kind of `[AppliesTo()]` that would filter this to only apply to .NET projects? I'm assuming our incremental builds code shouldn't apply to `.vcxproj` anyway.

Reply:
This is a great question and I wondered the same but didn't manage to work out how that'd work while putting this PR together at the end of my day yesterday. This might be a good topic for tomorrow's CPS meeting. If there's a better way, I will file a follow up PR.

The relevant line of code is here: https://github.com/dotnet/project-system/blob/3ce139b4711f3eb61964749b2ceb1e23d6796019/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Build/Diagnostics/IncrementalBuildFailureDetector.cs#L121

Reply:
I only asked this because I learned what `AppliesTo` meant from Tom's presentation this morning at the architectural meeting. 😝 You should check out the recording if you haven't already.

Reply:
You want to tag any .NET-specific exports with: `[AppliesTo(ProjectCapability.DotNet)]`. If they are actually specific to C# and VB, then `[AppliesTo(ProjectCapability.CSharpOrVisualBasic)]`.

Reply:
We would just duplicate the capabilities from our IUpToDateCheckProvider, which also excludes shared projects.